### ISSUE TITLE
FinalLocalVariable property to only validate default assignment

### ISSUE THREAD

**Issue by delanym** [checkstyle/checkstyle#17644]
Currently the https://checkstyle.sourceforge.io/checks/coding/finallocalvariable.html
check is quite heavy for a codebase that has a lot of reassignment.

I propose a more relaxed alternative that can be applied to prioritise refactoring of variables that are declared with a default which is possibly reassigned. For example, to avoid reassignment, `methodA` should rather be refactored to `methodB`

```java
public class FinalLocalVariableVariation {

  String methodA(List<String> quota){
    String title = "Full quota";

    if (quota.isEmpty()) {
      title = quota.getFirst();
    }

    return gist(title);
  }

  String methodB(List<String> quota){
    String title;

    if (quota.isEmpty()) {
      title = quota.getFirst();
    } else {
      title = "Full quota";
    }

    return gist(title);
  }

  String gist(String title) {return "123";}
  String gist(String title, List<String> stores) {return "321";}
```

This might not look so bad by itself, but when multiple variables are assigned and their defaults are handled in different ways the readability suffers

```java
public class FinalLocalVariableVariation {

  String methodC(List<String> quota){
    String title = "Full quota";
    List<String> stores;

    if (quota.isEmpty()) {
      title = quota.getFirst();
      stores = List.of("Full");
    } else {
      stores = null;
    }

    return gist(title, stores);
  }

  String methodD(List<String> quota){
    String title;
    List<String> stores;

    if (quota.isEmpty()) {
      title = quota.getFirst();
      stores = List.of("Full");
    } else {
      title = "Full quota";
      stores = null;
    }

    return gist(title, stores);
  }

  String gist(String title) {return "123";}
  String gist(String title, List<String> stores) {return "321";}

}
```
When the assignment is aligned like this, it becomes more obvious to take the refactoring further, if desired, and convert to ternary expressions:

```java
  String methodE(List<String> quota){
    return quota.isEmpty()
        ? gist(quota.getFirst(), List.of("Full"))
        : gist("Full quota", null);
  }
```

I would call the new property for this `OnlyValidateDefaultAssignment`
and when it is set to true the check would follow this logic:

1. Identify a variable that gets reassigned.
2. Confirm that only one reassignment is made.
3. Confirm the reassignment is part of a condition that has or could have a single compliment. For example, if the reassignment is within an if, is there a single else? Or if within an else, is there a single if? The same for an ternary expression or a switch (although switches should never have only 2 cases).
4. Mark as failing

**Comment by romani** [comment]
Please follow issue template, all java snippet should have config of checkstyle and output of CLI to show behavior

**Comment by delanym** [comment]
```xml
<module name="Checker">
  <module name="TreeWalker">
    <module name="FinalLocalVariable">
      <property name="onlyValidateDefaultAssignment" value="true"/>
    </module>
  </module>
</module>
```

How can I show the behaviour if it doesn't exist yet?

**Comment by romani** [comment]
Use config without imaginary property to show violations that you would like to avoid
